src/feedback_forensics/app/data/handler.py

```diff
@@ -0,0 +1,447 @@
+        Returns:
+            str: The path to the dataset.
+        """
+
+        if dataset_name not in self.avail_datasets:
+            raise ValueError(
+                f"Dataset {dataset_name} not found in avail_datasets ({self.avail_datasets})"
+            )
+
+        dataset_config = self.avail_datasets[dataset_name]
+        return dataset_config.path
+
+    def load_data_from_name(self, name: str):
+        """Load data from a given dataset name."""
+
+        assert self.avail_datasets is not None, (
+            "Avail_datasets must be provided to load data from name. "
+            "Use load_data_from_path instead if you have a path."
+        )
+
+        logger.info(f"Loading data from name: {name}")
+        self._data_path = self._get_data_path(name)
+        self.load_data_from_path(self._data_path)
+
+    def load_data_from_path(self, dataset_path: str | Path):
+        """Load data from a given path."""
+        dataset_path = Path(dataset_path)
+
+        base_votes_dict = get_votes_dict(dataset_path, cache=self.cache)
+        votes_dict = add_virtual_annotators(
+            base_votes_dict,
+            cache=self.cache,
+            dataset_cache_key=dataset_path,
+            reference_models=[],
+            target_models=[],
+        )
+
+        self.load_from_votes_dict(votes_dict)
+        logger.info(f"Loaded data from path: {dataset_path}")
+
+    def load_from_votes_dict(self, votes_dict: dict):
+        """Load data from a given votes_dict."""
+        self._df = votes_dict["df"]
+        self._annotator_metadata = votes_dict["annotator_metadata"]
+        self._reference_annotator_col = votes_dict["reference_annotator_col"]
+        self._shown_annotator_rows = votes_dict["shown_annotator_rows"]
+
+    def set_visible_annotator_rows(self, annotator_rows_keys: list[str]):
+        """Change the visible annotator rows for the votes_dict."""
+        if not annotator_rows_keys or len(annotator_rows_keys) == 0:
+            logger.warning(
+                "No annotator rows keys provided. Not changing annotator rows."
+            )
+            return
+
+        self._shown_annotator_rows = annotator_rows_keys
+
+    def compute_overall_metrics(self):
+        """Compute the overall metrics for the votes_dict."""
+        return get_overall_metrics(self.df, self.reference_annotator_col)
+
+    def compute_annotator_metrics(self):
+        """Compute the annotator metrics for the votes_dict."""
+        return compute_annotator_metrics(
+            votes_df=self.df,
+            annotator_metadata=self.annotator_metadata,
+            annotator_cols=self.shown_annotator_rows,
+            ref_annotator_col=self.reference_annotator_col,
+        )
+
+
+class DatasetHandler:
+    """Class to handle dataset operations of multi-column annotation datasets.
+
+    A dataset consists of one or multiple annotation columns, represented by
+    ColumnHandler objects. Each column can either be a different annotator or
```

Comment:
In AnnotatedPairs, annotators are first-class objects and unrelated to columns. Should we adopt this here too? I find it a bit confusing to entangle the concept with columns. Ideally the column -> annotator conversion should be done once (maybe even in a preprocessing step), and then we can forget about it and just deal with annotators. Or would you say that this *is* the preprocessing step?

Reply:
Yeah I think we may want to simplify this in the future, the current implementation basically just mirrors what we had before with the votes_dicts - which really works on columns as separate entities (whether annotator-based, separate datasets or data subsets). I would leave it as is for this PR, but ideally the current level of abstraction will make it easier to further adapt it in the future. 

Or do you think there would be a relatively simple way to implement this for this PR?